src/index.ts

```diff
@@ -265,12 +265,25 @@ export class Paginator {
     const callback = parsedArguments.callback!;
     if (parsedArguments.autoPaginate) {
       const results = new Array<{}>();
-      paginator.runAsStream_(parsedArguments, originalMethod)
-          .on('error', callback)
-          .on('data', (data: {}) => results.push(data))
-          .on('end', () => callback(null, results));
+      if (callback) {
+        paginator.runAsStream_(parsedArguments, originalMethod)
+            .on('error', callback)
+            .on('data', (data: {}) => results.push(data))
+            .on('end', () => callback(null, results));
+      } else {
+        return new Promise((resolve, reject) => {
+          paginator.runAsStream_(parsedArguments, originalMethod)
```

Comment:
Isnt it better to create this stream before if condition? to avoid duplicate code?

Reply:
Technically you can just convert this in to lambda which creates two callback and then call that lambda from if condition to avoid this whole duplicate stuff probably.

Reply:
the following `returns` differ
```ts
.on('end', () => callback(null, results));
.on('end', () => resolve([results]));
```
`promise expects an `Array` of results, where callback expects `null` for error and `results` as a second parameter.